File: Metode-Avansate-de-Programare/lab2/model/SortingTask.java
package sem1_2.model;

import java.util.ArrayList;

public class SortingTask extends Task {
    private ArrayList<Integer> toSort;
    private AbstractSorter sorter;

    public SortingTask(String id, String description, ArrayList<Integer> toSort, AbstractSorter sorter) {
        super(id, description);
        this.toSort = toSort;
        this.sorter = sorter;
    }

    public ArrayList<Integer> getToSort() {
        return toSort;
    }

    public void setToSort(ArrayList<Integer> toSort) {
        this.toSort = toSort;
    }

    public void execute(){
        System.out.println(sorter.sort(toSort));
    }
}

File: Metode-Avansate-de-Programare/lab11/lab11/SortingTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab11;

public class SortingTask : Task
{
    private int[] vector;
    private AbstractSorter sorter;

    public SortingTask(string id, string description, int[] vector, AbstractSorter sorter) : base(id, description)
    {
        this.vector = vector;
        this.sorter = sorter;
    }

    public override void execute()
    {
        sorter.sort(ref vector);
        Console.WriteLine("Vector after sorting :");
        foreach (int i in vector) Console.WriteLine(i);
    }
}